includes/REST_API/Stories_Base_Controller.php

```diff
@@ -68,27 +69,200 @@ public function __construct( $post_type ) {
+			if ( isset( $registered[ $api_param ], $request[ $api_param ] ) ) {
+				$args[ $wp_param ] = $request[ $api_param ];
+			}
+		}
+
+		// Check for & assign any parameters which require special handling or setting.
+		$args['date_query'] = [];
+
+		// Set before into date query. Date query must be specified as an array of an array.
+		if ( isset( $registered['before'], $request['before'] ) ) {
+			$args['date_query'][0]['before'] = $request['before'];
+		}
+
+		// Set after into date query. Date query must be specified as an array of an array.
+		if ( isset( $registered['after'], $request['after'] ) ) {
+			$args['date_query'][0]['after'] = $request['after'];
+		}
+
+		// Ensure our per_page parameter overrides any provided posts_per_page filter.
+		if ( isset( $registered['per_page'] ) ) {
+			$args['posts_per_page'] = $request['per_page'];
+		}
+
+		// Force the post_type argument, since it's not a user input variable.
+		$args['post_type'] = $this->post_type;
+
+		/**
+		 * Filters the query arguments for a request.
+		 *
+		 * Enables adding extra arguments or setting defaults for a post collection request.
+		 *
+		 * @link https://developer.wordpress.org/reference/classes/wp_query/
+		 *
+		 * @since 1.0.0
+		 *
+		 * @param array           $args    Key value array of query var to query value.
+		 * @param WP_REST_Request $request The request used.
+		 */
+		$args       = apply_filters( "rest_{$this->post_type}_query", $args, $request );
+		$query_args = $this->prepare_items_query( $args, $request );
+
+		$taxonomies = wp_list_filter( get_object_taxonomies( $this->post_type, 'objects' ), [ 'show_in_rest' => true ] );
+
+		if ( ! empty( $request['tax_relation'] ) ) {
+			$query_args['tax_query'] = [ 'relation' => $request['tax_relation'] ]; // phpcs:ignore WordPress.DB.SlowDBQuery.slow_db_query_tax_query
+		}
+
+		foreach ( $taxonomies as $taxonomy ) {
+			$base        = ! empty( $taxonomy->rest_base ) ? $taxonomy->rest_base : $taxonomy->name;
+			$tax_exclude = $base . '_exclude';
+
+			if ( ! empty( $request[ $base ] ) ) {
+				$query_args['tax_query'][] = [
+					'taxonomy'         => $taxonomy->name,
+					'field'            => 'term_id',
+					'terms'            => $request[ $base ],
+					'include_children' => false,
+				];
+			}
+
+			if ( ! empty( $request[ $tax_exclude ] ) ) {
+				$query_args['tax_query'][] = [
+					'taxonomy'         => $taxonomy->name,
+					'field'            => 'term_id',
+					'terms'            => $request[ $tax_exclude ],
+					'include_children' => false,
+					'operator'         => 'NOT IN',
+				];
+			}
+		}
+
+		// Add counts for other statuses.
+		$statuses = [
+			'all'     => [ 'publish', 'draft', 'future', 'private' ],
+			'publish' => 'publish',
+			'future'  => 'future',
+			'draft'   => 'draft',
+			'private' => 'private',
+		];
+
+		$statuses_count = [];
+
+		// Strip down query for speed.
+		$query_args['fields']                 = 'ids';
+		$query_args['posts_per_page']         = 1;
```

Comment:
Probably this shouldn't belong to the `Stories_Base_Controller`? Generally, there seem to be things here that are not necessary for other post types, e.g. the statuses, taxonomies (for now at least), etc. Thoughts?

Reply:
It's basically all copy paste from `\WP_REST_Posts_Controller::get_items()`. Stuff that the REST API supports for all post types out of the box.

Reply:
Yep, saw it was basically just moved over but isn't this logic unsuitable for the Post Templates?

Reply:
`$query_args['posts_per_page']         = 1;` this would influence it though, no?

Reply:
That's used for the `X-WP-TotalByStatus` part below. Not to be confused with `$args`

That `X-WP-TotalByStatus` part should probably only be in `Stories_Controller` as we only need it for stories.

^ @spacedmonkey

Reply:
The test failure here https://github.com/google/web-stories-wp/runs/2555275031#step:15:30

This because the rest the request object is being double enveloped. I another way around this would be a new class.